packages/patternfly-4/react-core/src/components/Switch/Switch.tsx

```diff
@@ -62,14 +62,14 @@ class Switch extends React.Component<SwitchProps & InjectedOuiaProps> {
         })}
       >
         <input
+          {...props}
           id={this.id}
           className={css(styles.switchInput)}
           type="checkbox"
           onChange={event => onChange(event.target.checked, event)}
           checked={isChecked}
           disabled={isDisabled}
           aria-labelledby={isAriaLabelledBy ? `${this.id}-on` : null}
-          {...props}
```

Comment:
I don't know that we want all props added to the input? For example, there are props (i.e., className, ouiaId, etc.) that are only applied to the label.

Probably should be using `rest` instead of all `props`? Then, we can destructure the `id` so that it does not override anything else.

This:
`const { className, label, labelOff, isChecked, isDisabled, onChange, ouiaContext, ouiaId, ...props } = this.props;`

becomes:
`const { className, label, labelOff, id, isChecked, isDisabled, onChange, ouiaContext, ouiaId, ...rest } = this.props;`

Finally:       
```
<input
  id={this.id}
  className={css(styles.switchInput)}
  type="checkbox"
  onChange={event => onChange(event.target.checked, event)}
  checked={isChecked}
  disabled={isDisabled}
  aria-labelledby={isAriaLabelledBy ? `${this.id}-on` : null}
  {...rest} // <<< This should be last
/>
```